app/test/v1/test_product.py

```diff
@@ -4,33 +4,95 @@
 
 
 class TestProduct(TestCase):
+
     def setUp(self):
         self.app = create_app('testing')
         self.client = self.app.test_client()
+        self.add_user()
+        self.login_users()
         self.product = {
             "name": "sugar",
             "cost": 10,
             "amount": 3
         }
 
+    def add_user(self):
+        user_login = self.client.post(
```

Comment:
Handle this part of user authentication in a separate file so that you don't repeat it in the other `TestSales` file, instead you just import it and use it in your test files